cardano_node_tests/tests/test_cli.py

```diff
@@ -78,6 +78,48 @@ def test_whole_utxo(self, cluster: clusterlib.ClusterLib):
             ]
         )
 
+    @allure.link(helpers.get_vcs_link())
+    @pytest.mark.skipif(
+        bool(configuration.TX_ERA),
+        reason="different TX eras doesn't affect this test, pointless to run",
+    )
+    def test_utxo_txin(self, cluster: clusterlib.ClusterLib):
+        """Check that it is possible to filter UTxO based on transaction input on local cluster."""
+        if cluster.protocol != clusterlib.Protocols.CARDANO:
+            pytest.skip("runs on cluster in full cardano mode")
+        utxos = cluster.cli(
+            [
+                "query",
+                "utxo",
+                "--whole-utxo",
+                *cluster.magic_args,
+            ]
+        )
+
+        utxo = utxos[0].splitlines()[2].decode("UTF-8").split()
+        if not utxo:
+            raise AssertionError("No data returned from 'cardano-cli query utxo --whole-utxo'.")
+
+        tx_hash, tx_ix, _, _, _, _ = utxo
+
```

Comment:
Good start!
I propose to add a `payment_addr` fixture that creates & funds one payment address, similarly as it is done e.g. in https://github.com/input-output-hk/cardano-node-tests/blob/73d503e55596bf6c9ee86f39735ebc8f267b9ca6/cardano_node_tests/tests/test_transactions.py#L113
This way you could then call
```
utxo_rec = cluster.get_utxo(payment_addr)[0]
utxo_txin = f"{utxo_rec.utxo_hash}#{utxo_rec.utxo_ix}"
```
and use that in the query below. The advantage is that the test can run also on testnets where `--whole-utxo` would return too many items.

Eventually we want to add the support for `query utxo --tx-in` to [cardano-clusterlib](https://github.com/input-output-hk/cardano-clusterlib-py) and use it from there, instead of calling it directly like this, as this functionality can be useful in other places as well. But that can be another step.

Reply:
Thanks for the review Martin. I will apply your suggestions.